libs/net/template/Models/BaseTemplateModel.cs

```diff
@@ -10,6 +10,16 @@ namespace TNO.TemplateEngine.Models;
 public abstract class BaseTemplateModel<T> : RazorEngineTemplateBase
 {
     #region Properties
+    /// <summary>
+    /// get/set - The primary key to the report.
+    /// </summary>
+    public int ReportId { get; set; }
```

Comment:
Including these primary keys provides a way for reports to include a link to themselves.